barretenberg/cpp/src/barretenberg/protogalaxy/prover_verifier_shared.hpp

```diff
@@ -1,20 +1,57 @@
 #pragma once
```

Comment:
Moved everything to the header to re-use by the recursive verifier. Could try doing the same with explicit template instantiation, but seemingly would be messier.

Reply:
This is bad practice but since the functions are not used very much and they're small we can live with it.